src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -655,21 +665,58 @@ public void ProduceContinue()
                 }
             }
 
-            if (_keepAlive == false && hasConnection == false && HttpVersion == "HTTP/1.1")
+            if (_keepAlive == false && _responseHeaders.HasConnection == false && _httpVersion == HttpVersionType.Http1_1)
             {
-                writer.Write("Connection: close\r\n\r\n");
+                _responseHeaders.SetRawConnection("close", _bytesConnectionClose);
             }
-            else if (_keepAlive && hasConnection == false && HttpVersion == "HTTP/1.0")
+            else if (_keepAlive && _responseHeaders.HasConnection == false && _httpVersion == HttpVersionType.Http1_0)
             {
-                writer.Write("Connection: keep-alive\r\n\r\n");
+                _responseHeaders.SetRawConnection("keep-alive", _bytesConnectionKeepAlive);
+            }
+
+            end.CopyFromAscii(_httpVersion == HttpVersionType.Http1_1 ? _bytesHttpVersion1_1 : _bytesHttpVersion1_0);
+            end.CopyFromAscii(statusBytes);
+            _responseHeaders.CopyTo(ref end);
+            end.CopyFromAscii(_bytesEndHeaders, 0, _bytesEndHeaders.Length);
+
+            // TODO: change this to SocketOutput.ProduceStart/ProduceComplete once that change is made 
+            var scan = begin.Block;
```

Comment:
Do you mean the whole PR? This bit passes the individual ArraySegements into WriteAsync as it can't currently take a set of memoryblocks. Will change after @halter73's change to just pass the memory chain in (I believe?)